gridcapa-cse-valid-app/src/main/java/com/farao_community/farao/cse_valid/app/rao/CseValidRaoValidator.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ *  This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.cse_valid.app.rao;
+
+import com.farao_community.farao.cse_valid.api.resource.CseValidRequest;
+import com.farao_community.farao.cse_valid.api.resource.ProcessType;
+import com.farao_community.farao.cse_valid.app.FileExporter;
+import com.farao_community.farao.cse_valid.app.FileImporter;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_creation.creator.cse.CseCrac;
+import com.farao_community.farao.data.rao_result_api.OptimizationState;
+import com.farao_community.farao.data.rao_result_api.RaoResult;
+import com.farao_community.farao.rao_runner.api.resource.RaoRequest;
+import com.farao_community.farao.rao_runner.api.resource.RaoResponse;
+import com.farao_community.farao.rao_runner.starter.RaoRunnerClient;
+import com.powsybl.iidm.network.Network;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.time.OffsetDateTime;
+
+/**
+ * @author Oualid Aloui {@literal <oualid.aloui at rte-france.com>}
+ */
+
+@Component
+public class CseValidRaoValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CseValidRaoValidator.class);
+
+    private final FileImporter fileImporter;
+    private final FileExporter fileExporter;
+    private final RaoRunnerClient raoRunnerClient;
+    private int variantCounter = 0;
+
+    public CseValidRaoValidator(FileImporter fileImporter, FileExporter fileExporter, RaoRunnerClient raoRunnerClient) {
+        this.fileImporter = fileImporter;
+        this.fileExporter = fileExporter;
+        this.raoRunnerClient = raoRunnerClient;
+    }
+
+    public boolean isNetworkSecure(Network network, CseValidRequest cseValidRequest, String cracUrl) {
+        ProcessType processType = cseValidRequest.getProcessType();
+        OffsetDateTime processTargetDateTime = cseValidRequest.getTimestamp();
+        String requestId = cseValidRequest.getId();
+
+        String jsonCracUrl = getJsonCracUrl(cseValidRequest, network, cracUrl);
+        String raoParametersURL = fileExporter.saveRaoParameters(processTargetDateTime, processType);
+        String scaledNetworkDirPath = generateScaledNetworkDirPath(network, processTargetDateTime, processType);
+        String scaledNetworkName = network.getNameOrId() + ".xiidm";
+        String networkFiledUrl = fileExporter.saveNetworkInArtifact(network, scaledNetworkDirPath + scaledNetworkName, "", processTargetDateTime, processType);
+
+        RaoRequest raoRequest = new RaoRequest(requestId, networkFiledUrl, jsonCracUrl, raoParametersURL, "CSE/VALID/");
```

Comment:
Setting "CSE/VALID/ means all the files generated by the RAO i.e. RaoResult, network with preventive remedial actions for example will be stored directly in this directory, which is kind of the root for all CSE/VALID/ process files, I'm not sure that's what we want. Think properly where to store those files on MinIO : the initially shifted network and the sub results of the following RAO. Compared to what is done so far it could be something more like CSE/VALID/D2CC/2023/01/03/07_30/ARTIFACTS/INITIAL_SHIFT (of course with the date that matches the task date)